### PR TITLE
codegen new types from sigstore protobufs

### PR DESCRIPTION
**Pull request by bdehamer**
Signed-off-by: Brian DeHamer <bdehamer@github.com>

#### Summary
Rejiggering the code generation for the Sigstore protobufs. The Sigstore [protobufs](https://github.com/sigstore/protobuf-specs) now include more than just the definition for the bundle format, so putting the generated types into a directory named "bundle" no longer makes sense. The generated code is now placed in a directory called "sigstore".

Also, some of the new protobufs reference 3rd-party protos that were not present in the Docker image I was using to wrap the `protoc` CLI. I've switched to using the `namely/protoc-all` base image was already contains a number of commonly-referenced plugins and protobuf definitions. This is the same [base image](https://github.com/sigstore/protobuf-specs/blob/main/Dockerfile#L1) used by the Sigstore `protobuf-spec` project.

### CODE REVIEW

File: hack/images/protoc/Dockerfile
Comment (feelepxyz) [inline hack/images/protoc/Dockerfile:2]: Nice 👍

**Review by feelepxyz — APPROVED** [review]
Will this cause any issues because of the changed bundle format and not integrating these changes? https://github.com/sigstore/sigstore-js/pull/212

**Comment by bdehamer** [comment]
> Will this cause any issues because of the changed bundle format and not integrating these changes? https://github.com/sigstore/sigstore-js/pull/212

For now these types will live alongside the older "bundle" types. I wanted to be able to use some of the verification-related types without being blocked by the breaking bundle changes.

After we merge #212 I'll fold all this together so that we have a single copy of the generated code.

**Pull request merged**
